app/src/androidTest/java/com/sdp/movemeet/view/chat/ChatActivitySendImageTest.java

```diff
@@ -0,0 +1,71 @@
+package com.sdp.movemeet.view.chat;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.view.profile.EditProfileActivity;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import static junit.framework.Assert.assertTrue;
+
+public class ChatActivitySendImageTest {
+
+    private StorageReference storageReference;
+    @Rule
+    public IntentsTestRule<ChatActivity> activityRule = new IntentsTestRule<>(ChatActivity.class);
+
+    @Before
+    public void setUp() {
+        // Setting up Firebase Storage
+        storageReference = FirebaseStorage.getInstance().getReference();
+
+        // Saving the mocked picked image
+        savePickedImage();
+    }
+
+    @Test
+    public void sendImageViaChat() {
+        File dir = activityRule.getActivity().getExternalCacheDir();
+        File file = new File(dir.getPath(), "pickImageResult.jpeg");
+        Uri imageUri = Uri.fromFile(file);
+        String fullNameString = "Anthony";
+        String userId = "xzkBeYkTY7ccFIYY6F9OHDuup8I2";
+        activityRule.getActivity().createTempMessage(imageUri, fullNameString, userId);
+    }
+
+
+
+    private void savePickedImage() {
+        Bitmap bm = BitmapFactory.decodeResource(activityRule.getActivity().getResources(), R.drawable.run_man);
+        assertTrue(bm != null);
+        File dir = activityRule.getActivity().getExternalCacheDir();
+        File file = new File(dir.getPath(), "pickImageResult.jpeg");
```

Comment:
Will this kind of test work on CI? I'm not sure we can open files we haven't generated first

Reply:
This image to be put into this file is actually coming from the app itself (it is the default profile picture of the user). I wrote this test two weeks ago and, so far, it seems hence to run on CI.

Reply:
Oh great! Don't mind me then

Reply:
In fact you were right. The way I did it in `EditProfileActivityImagePickerTest` passes on CI, but not the one here in `ChatActivitySendImageTest`... would you have any idea why this is so?

Reply:
I'm not quite sure, but you might be able to solve the issue by using mocks. If you're available during the week we could discuss it over Zoom

Reply:
Oh yes! That would be so nice! Thank you for this, we will organize!